plugin/pkg/scheduler/scheduler.go

```diff
@@ -89,6 +92,12 @@ func (s *Scheduler) Run() {
 
 func (s *Scheduler) scheduleOne() {
 	pod := s.config.NextPod()
+
+	// If the assumed pods store contains this pod there is a conflict,
+	// since the store can't contain a pod without a host, and NextPod
+	// only gives us pods without hosts.
+	s.config.Modeler.ForgetAssumedPod(pod)
```

Comment:
Multiple questions:

A) Yeah, I understand that-- but suppose the pod was deleted because of a bad assignment. Scheduler is going to put it right back where it came from unless we do something to stop that.

B) Is this to fix the problem where pods stick around in the assumed pods list? I just observed this at @vishh's desk and I think there must be some other problem, because controller-manager creates a pod with a different name every time. So I don't expect this change to resolve that problem.